assets/js/blocks/attribute-filter/block.js

```diff
@@ -268,29 +268,36 @@ const AttributeFilterBlock = ( {
 		[ pageUrl, attributeObject?.taxonomy ]
 	);
 
+	const onClickSubmit = ( isChecked ) => {
+		const query = updateAttributeFilter(
+			productAttributesQuery,
+			setProductAttributesQuery,
+			attributeObject,
+			getSelectedTerms( isChecked ),
+			blockAttributes.queryType === 'or' ? 'in' : 'and'
+		);
+
+		// This is for PHP rendered template filtering only.
+		if ( filteringForPhpTemplate ) {
+			redirectPageForPhpTemplate( query, isChecked.length === 0 );
+		}
+	};
```

Comment:
Using `onClickSubmit` without `useCallback` makes the `<FilterSubmitButton/>` re-render every time an attribute is checked. But I don't think it affects the performance noticeably.

Reply:
Perhaps it can be updated in this PR https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6333 to use `useCallback` unless we can see a reason why it cannot?

Reply:
@tjcafferkey Perhaps investigating why it works with two Filter Products by Attribute blocks gives us some clue?

Reply:
FYI, with one Filter Products by Attribute block, at the time the filter button is clicked, `hasSetPhpFilterDefaults` is `false`, it means that `setHasSetPhpFilterDefaults` hasn't been called yet.

Reply:
The issue with the current codebase is, that we're calling `redirectPageForPhpTemplate()` in two places, and clicking the filter button can trigger both function calls (when conditions meet). This makes the behavior of the Filter button unpredictable and hard to debug.

Reply:
> Using `onClickSubmit` without `useCallback` makes the `<FilterSubmitButton/>` re-render every time an attribute is checked. But I don't think it affects the performance noticeably.

Wasn't that already happening? The onclick `prop` was different all the time because we were passing an arrow function that was created on every render.

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/2fb5c28757099952e236731437eadc0d58a610c1/assets/js/blocks/attribute-filter/block.js#L591

Reply:
> Wasn't that already happening?

I'm not sure why it didn't happen when I test it here https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6332#discussion_r859569876.

> The onclick prop was different all the time because we were passing an arrow function that was created on every render.

You're right, a closure is passed to `onClick` so it's recreated on every render. I'm sorry for the early conclusion! : (